src/plots/plots.js

```diff
@@ -702,7 +703,17 @@ function getFormatter(formatObj, separators) {
     formatObj.thousands = separators.charAt(1);
 
     return {
-        numberFormat: d3.locale(formatObj).numberFormat,
+        numberFormat: function(formatStr) {
+            try {
+                formatStr = formatLocale(formatObj).format(
+                    Lib.adjustFormat(formatStr)
+                );
+            } catch(e) {
+                return Lib.noFormat;
```

Comment:
Is this catching invalid format strings? Do we want to `lib.warn` when this happens, to help people fix it?

Reply:
Done in https://github.com/plotly/plotly.js/pull/5125/commits/878034b5ae2e78d5b855d2267325d330e8669715